mod-login-saml/src/main/resources/domain/mod-login-saml/features/loginSaml.feature

```diff
@@ -3,11 +3,143 @@ Feature: Login SAML
   Background:
     * url baseUrl
     * callonce login testUser
-    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain' }
-
-  Scenario: SAML Check
-     Given path 'saml/check'
-     And header x-okapi-tenant = testTenant
-     And header x-okapi-token = '#(okapitoken)'
-     When method GET
-     Then status 200
+    * configure headers = { 'Content-Type': 'application/json', 'x-okapi-tenant': #(testTenant), 'x-okapi-token': '#(okapitoken)', 'Accept': 'application/json, text/plain' }
+
+  Scenario: Check endpoint missing okapi url header active is false
+    Given path 'saml/check'
+    When method GET
+    Then status 200
+    # A lot of the Rest Assured tests use the matchesJsonSchemaInClasspath method to validate JSON schema.
+    # Karate has a different opinion about how to validate JSON schema. See https://github.com/intuit/karate#schema-validation
+    # Here the only part of the schema returned is one property so about all we can check is the data type of that property
+    # (this is karate's 'schema validation') and the value.
+    Then match response ==
+    """
+    {
+      active: '#boolean'
+    }
+    """
+    Then match response.active == false
+
+  # TODO In the rest assured tests this is 400 active true. But here it is active false 200. Which is correct?
```

Comment:
I do not see rest assured tests returns `400 active true`. Are you referring to this [code](https://github.com/folio-org/mod-login-saml/blob/v2.2.0/src/test/java/org/folio/rest/impl/SamlAPITest.java#L133-L164)? It returns 400 only when `x-okapi-tenant` header is missing.

Reply:
Yes that's right. That's a typo in my comment. It should read ```200 active true```. If I run the test that is [commented out](https://github.com/folio-org/folio-integration-tests/blob/28fbcf919a1bf87cdbed0bc1c56dd257f804af5e/mod-login-saml/src/main/resources/domain/mod-login-saml/features/loginSaml.feature#L28) I get this result, which I think I should instead be ```active: true``` to match the rest assured test.

```
17:28:16.260 request:
1 > GET https://folio-testing-okapi.dev.folio.org:443/saml/check
1 > Accept: application/json, text/plain
1 > Accept-Encoding: gzip,deflate
1 > Connection: Keep-Alive
1 > Content-Type: application/json
1 > Host: folio-testing-okapi.dev.folio.org:443
1 > User-Agent: Apache-HttpClient/4.5.12 (Java/11.0.10)
1 > x-okapi-tenant: testenant6106266363388642304
1 > x-okapi-token: eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ0ZXN0LXVzZXIiLCJ1c2VyX2lkIjoiMDAwMDAwMDAtMTExMS01NTU1LTk5OTktOTk5OTk5OTk5OTkyIiwiaWF0IjoxNjIzNDQ2ODk2LCJ0ZW5hbnQiOiJ0ZXN0ZW5hbnQ2MTA2MjY2MzYzMzg4NjQyMzA0In0.IhvX1lVXsdrLXjvHdvDEmF03M3ujhThkCwriegZV9Yc
1 > x-okapi-url: http://localhost:9130

17:28:16.398 response time in milliseconds: 137.23
1 < 200
1 < Connection: keep-alive
1 < Content-Type: application/json
1 < Date: Fri, 11 Jun 2021 21:28:16 GMT
1 < Transfer-Encoding: chunked
1 < X-Okapi-Trace: GET mod-login-saml-2.2.1-SNAPSHOT.70 http://10.36.1.135:9146/saml/check : 200 14735us
1 < vary: origin
{
  "active": false
}
```
Maybe this is because there is no IdP configured. I think it's this code [here](https://github.com/folio-org/mod-login-saml/blob/d8e202e0a307d12b2080dd5b76a7afd59463c8d9/src/main/java/org/folio/rest/impl/SamlAPI.java#L556) and probably [this](https://github.com/folio-org/mod-login-saml/blob/d8e202e0a307d12b2080dd5b76a7afd59463c8d9/src/main/java/org/folio/config/SamlClientLoader.java#L64) that are causing ```active: false```.

Part of the problem with these karate tests is that -- at least I haven't figured it out yet -- is they're not that easy to debug. So I'm not actually able to see if [this line](https://github.com/folio-org/mod-login-saml/blob/d8e202e0a307d12b2080dd5b76a7afd59463c8d9/src/main/java/org/folio/config/SamlClientLoader.java#L64)  is being hit by placing a breakpoint or even console logging because the module is being pulled in through maven. Probably there's a way to have maven point a local mod-login-saml project. But anyway you see what I mean...

Let me know what you think. It would be nice to figure out why this test isn't working.

Reply:
The result of `200 active false` is as expected because https://folio-testing.dev.folio.org/ does not have IdP configured.

Reply:
Hongwei just noticing I didn't reply back to this. It is good to have this confirmed. Thanks for taking a look at it.